### PR TITLE
[TGUI] Particle Accelerator

### PR DESCRIPTION
**Pull request by Burzah**
## What Does This PR Do
This PR adds a new TGUI interface for the Particle Accelerator. Parts of this PR were loosely ported from the NSV/Bee code base. 

## Why It's Good For The Game
Improves player experience by providing a modern UI inline with other UIs currently in game. This also strikes another TGUI conversion off of https://github.com/ParadiseSS13/Paradise/issues/14968

### Before:
RzoTKw5TRO
### After:
dreamseeker_WwdujqpBlr

## Testing
Built the particle accelerator and tested the connect feature.
Tested the on and off switch.
Tested the strength adjustment.

## Changelog
:cl:
add: Added new TGUI for the Particle Accelerator
del: Removed old UI
/:cl:

### CODE REVIEW

File: code/modules/power/engines/singularity/particle_accelerator/particle_control.dm [hunk 1]
```diff
@@ -247,36 +213,42 @@
...
+/obj/machinery/particle_accelerator/control_box/ui_data(mob/user)
+	var/list/data = list()
+	data["assembled"] = assembled
+	data["power"] = active
+	data["strength"] = strength
+	return data
+
+/obj/machinery/particle_accelerator/control_box/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	if(..())
+		return
```
Comment (Contrabang) [inline code/modules/power/engines/singularity/particle_accelerator/particle_control.dm:232]: I think you should readd the variable check for `interface_control` here, that was in topic with the `to_chat(usr, "<span class='error'>ERROR: Request timed out. Check wire contacts.</span>")` line

File: code/modules/power/engines/singularity/particle_accelerator/particle_control.dm [hunk 2]
```diff
@@ -247,36 +213,42 @@
...
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "ParticleAccelerator", name, 350, 160, master_ui, state)
+		ui.open()
+
+/obj/machinery/particle_accelerator/control_box/ui_data(mob/user)
+	var/list/data = list()
+	data["assembled"] = assembled
+	data["power"] = active
+	data["strength"] = strength
```
Comment (Contrabang) [inline code/modules/power/engines/singularity/particle_accelerator/particle_control.dm:226]:
```suggestion
	data["strength"] = strength
	data["max_strength"] = strength_upper_limit
```

File: tgui/packages/tgui/interfaces/ParticleAccelerator.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                icon={power ? 'power-off' : 'times'}
+                content={power ? 'On' : 'Off'}
+                selected={power}
+                disabled={!assembled}
+                onClick={() => act('power')} />
+            </LabeledList.Item>
+            <LabeledList.Item label="Strength">
+              <Button
+                icon="backward"
+                disabled={!assembled}
```
Comment (Contrabang) [inline tgui/packages/tgui/interfaces/ParticleAccelerator.js:41]:
```suggestion
                disabled={!assembled || strength === 0}
```

File: tgui/packages/tgui/interfaces/ParticleAccelerator.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                onClick={() => act('remove_strength')} />
+              {' '}
+              {String(strength).padStart(1, '0')}
+              {' '}
+              <Button
+                icon="forward"
+                disabled={!assembled}
```
Comment (Contrabang) [inline tgui/packages/tgui/interfaces/ParticleAccelerator.js:48]:
```suggestion
                disabled={!assembled || strength === max_strength}
```

File: tgui/packages/tgui/interfaces/ParticleAccelerator.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+                onClick={() => act('power')} />
+            </LabeledList.Item>
+            <LabeledList.Item label="Strength">
+              <Button
+                icon="backward"
+                disabled={!assembled}
+                onClick={() => act('remove_strength')} />
+              {' '}
+              {String(strength).padStart(1, '0')}
+              {' '}
```
Comment (Contrabang) [inline tgui/packages/tgui/interfaces/ParticleAccelerator.js:45]: I'd say try converting this to a box, maybe something with 5 px padding or margins, try some negatives too to see how it looks
```suggestion
              <Box px="5px>
                  {strength}
              </Box>
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
I think adding the buttons being disabled when you can't press em would be a good feature

File: code/modules/power/engines/singularity/particle_accelerator/particle_control.dm (hunk 1 above)
Comment (Burzah) [reply code/modules/power/engines/singularity/particle_accelerator/particle_control.dm:232]: Oops. 
Resolved with [b777308](https://github.com/ParadiseSS13/Paradise/pull/21370/commits/b7773087a15a721ce43a60e6fe12af319006b2b3)

File: tgui/packages/tgui/interfaces/ParticleAccelerator.js (hunk 5 above)
Comment (Burzah) [reply tgui/packages/tgui/interfaces/ParticleAccelerator.js:45]: I am going to commit this and play around with it a bit more to see how it looks.

**Review by Sirryan2002 — APPROVED** [review]

**Pull request merged**